.github/workflows/build.yml

```diff
@@ -256,3 +256,48 @@ jobs:
           dev_tags: |
             docker.io/hashicorppreview/${{ env.repo }}:${{ env.dev_tag }}-ubi
             docker.io/hashicorppreview/${{ env.repo }}:${{ env.dev_tag }}-ubi-${{ github.sha }}
+
```

Comment:
Hey @claire-labry and @marianoasselborn 👋🏻 

I think this part needs your review because RelEng are the codeowners of our build configuration.

I did originally have these tests in a separate workflow, but I wanted to run them against the real release Docker image because we also bundle Envoy and go-discover.